dotnet/src/Extensions/TemplateEngine.Basic/Blocks/CodeBlock.cs

```diff
@@ -101,53 +101,25 @@ public async Task<string> RenderCodeAsync(SKContext context, CancellationToken c
 
     private async Task<string> RenderFunctionCallAsync(FunctionIdBlock fBlock, SKContext context)
     {
-        if (context.Functions == null)
-        {
-            throw new SKException("Function collection not found in the context");
-        }
-
-        if (!this.GetFunction(context.Functions!, fBlock, out ISKFunction? function))
-        {
-            var errorMsg = $"Function `{fBlock.Content}` not found";
-            this.Logger.LogError(errorMsg);
-            throw new SKException(errorMsg);
-        }
-
-        SKContext contextClone = context.Clone();
+        ContextVariables inputVariables = context.Variables.Clone();
```

Comment:
1. Please add a comment explaining the reason for the cloning.
2. nit: Consider putting the line into the `else` block to avoid double coning of context variables.